### PR TITLE
feat: add bucket migrate & sp exit query cli 

### PR DESCRIPTION
**Pull request by jingjunLi**
### Description

add bucket migrate & sp exit query cli.

### Rationale

NA

### Example

* bucket.migrate

```
./build/gnfd-sp query.bucket.migrate --config deployment/localup/local_env/sp1/config.toml
{
    "bucket_migrate":[
        {
            "bucket_id":1,
            "gvg_task":[
                {
                    "dest_gvg_id":2,
                    "src_gvg_id":1,
                    "status":2
                }
            ]
        }
    ],
    "self_sp_id":2
}
```

* sp.exit
```sh
./build/gnfd-sp query.sp.exit --config deployment/localup/local_env/sp2/config.toml
```

* src output
```
{
    "swap_out_src":[
        {
            "swap_out_key":"gvgIDList-1",
            "status":1,
            "successor_sp_id":8
        },
        {
            "swap_out_key":"familyID-2",
            "status":1,
            "successor_sp_id":1
        }
    ],
    "self_sp_id":2
}
```
* dest output
```
{
    "swap_out_dest":[
        {
            "swap_out_key":"gvgIDList-1",
            "successor_sp_id":8,
            "gvg_task":[
                {
                    "src_gvg_id":1,
                    "status":2
                }
            ]
        }
    ],
    "self_sp_id":8
}
```
### Changes

Notable changes: 
* NA
* ...

### CODE REVIEW

File: cmd/command/query.go [hunk 1]
```diff
@@ -126,6 +126,18 @@ var GetSegmentIntegrityCmd = &cli.Command{
...
 
+var QueryBucketMigrateCmd = &cli.Command{
+	Action: getBucketMigrateAction,
+	Name:   "query.bucket.migrate",
+	Usage:  "Query bucket migrate integrity hash and signature",
+	Flags: []cli.Flag{
+		utils.ConfigFileFlag,
+	},
+	Category: "QUERY COMMANDS",
+	Description: `The query.bucket.migrate command send rpc request to spdb
```
Comment (will-2012) [inline cmd/command/query.go:137]: Refine the description.

File: modular/manager/bucket_migrate_scheduler.go [hunk 2]
```diff
@@ -486,6 +486,26 @@ func (s *BucketMigrateScheduler) getExecutePlanByBucketID(bucketID uint64) (*Buc
 	}
 }
 
+func (s *BucketMigrateScheduler) listExecutePlan() (*gfspserver.GfSpQueryBucketMigrateResponse, error) {
+	var res gfspserver.GfSpQueryBucketMigrateResponse
+	var plans []*gfspserver.GfSpBucketMigrate
+	for _, executePlan := range s.executePlanIDMap {
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:492]: Need a read lock mutex.

File: proto/base/types/gfspserver/query_task.proto [hunk 3]
```diff
@@ -14,6 +15,48 @@ message GfSpQueryTasksResponse {
...
+
+message GfSpSPExit {
+  string bucket_name =2;
+  uint64 bucket_id = 3;
+  string migrate_status = 4;
+  repeated GfSpMigrateGVG gvg_task = 5;
+}
+
+message DestRunner {
+//  swap out ?
```
Comment (will-2012) [inline proto/base/types/gfspserver/query_task.proto:50]: swap out + swap out{gvg list}

File: modular/manager/bucket_migrate_scheduler.go [hunk 4]
```diff
@@ -486,6 +486,26 @@ func (s *BucketMigrateScheduler) getExecutePlanByBucketID(bucketID uint64) (*Buc
...
+			plan.GvgTask = append(plan.GvgTask, &gfspserver.GfSpMigrateGVG{
+				DestGvgId:            unit.destGVG.Id,
+				LastMigratedObjectId: unit.lastMigratedObjectID,
+				Status:               int32(unit.migrateStatus),
+			})
+		}
+		plans = append(plans, &plan)
+	}
+	res.BucketMigratePlan = plans
+	log.Debugw("listExecutePlan", "listExecutePlan", res)
```
Comment (sysvm) [inline modular/manager/bucket_migrate_scheduler.go:505]: use some useful descriptions.

File: modular/manager/sp_exit_scheduler.go [hunk 5]
```diff
@@ -6,6 +6,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/bnb-chain/greenfield-storage-provider/base/types/gfspserver"
```
Comment (sysvm) [inline modular/manager/sp_exit_scheduler.go:9]: sort imported pkg.

File: cmd/command/query.go [hunk 6]
```diff
@@ -340,3 +364,51 @@ func getSegmentIntegrityAction(ctx *cli.Context) error {
...
+	}
+	if ctx.IsSet(endpointFlag.Name) {
+		endpoint = ctx.String(endpointFlag.Name)
+	}
+	client := &gfspclient.GfSpClient{}
+	info, err := client.QueryBucketMigrate(context.Background(), endpoint)
+	if err != nil {
+		return err
+	}
+	fmt.Print(info)
```
Comment (sysvm) [inline cmd/command/query.go:387]: use `fmt.Println` may be better.

File: cmd/command/query.go [hunk 7]
```diff
@@ -340,3 +364,51 @@ func getSegmentIntegrityAction(ctx *cli.Context) error {
...
+	}
+	if ctx.IsSet(endpointFlag.Name) {
+		endpoint = ctx.String(endpointFlag.Name)
+	}
+	client := &gfspclient.GfSpClient{}
+	info, err := client.QuerySPExit(context.Background(), endpoint)
+	if err != nil {
+		return err
+	}
+	fmt.Print(info)
```
Comment (sysvm) [inline cmd/command/query.go:411]: ditto

File: modular/manager/manage_task.go [hunk 8]
```diff
@@ -7,6 +7,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/go-errors/errors"
+
```
Comment (sysvm) [inline modular/manager/manage_task.go:11]: remove meaningless blank line

**Review by will-2012 — APPROVED** [review]

**Review by sysvm — APPROVED** [review]
LGTM

**Pull request merged**